src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -910,5 +943,42 @@ private ICollection<ProjectWithInnerNodes> LoadProjects(IEnumerable<ProjectGraph
             // Just return the projects not the whole dictionary as it was just used to group the projects together
             return projects.Values;
         }
+
+        /// <summary>
+        /// It evaluates the project and returns true if the project has CentralPackageVersionManagement enabled.
+        /// </summary>
+        /// <param name="project">The <see cref="IMSBuildProject"/> for which the CentralPackageVersionManagement will be evaluated.</param>
+        /// <param name="projectStyle">The <see cref="ProjectStyle?"/>. Null is the project did not have a defined ProjectRestoreStyle property.</param>
+        /// <returns>True if the project has CentralPackageVersionManagement enabled and the project is PackageReference or the projectStyle is null.</returns>
+        internal static bool IsCentralVersionsManagementEnabled(IMSBuildProject project, ProjectStyle? projectStyle)
+        {
+            var cpvmEnabledProperty = StringComparer.OrdinalIgnoreCase.Equals(project.GetProperty("CentralPackageVersionsEnabled"), bool.TrueString);
+
+            return (cpvmEnabledProperty && (!projectStyle.HasValue || (projectStyle.Value == ProjectStyle.PackageReference)));
+        }      
```

Comment:
Minor performance improvement potential:  Check project style first, and only if it's a project type that CPVM allows, then get the property and compare the value. This code tries to get the MSBuild property value and do a string comparison, even when the project type isn't able to use CPVM.

Reply:
Updated.